### PR TITLE
Support batching of calls to `Documents.import_`

### PR DESCRIPTION
**Pull request by pratheekrebala**
## Change Summary
This PR allows batching of calls to the `Documents.import_` method and allows user to provide a generator to the `import_` function. This provides some QoL improvements when working with large number of documents.

Specifically, the changes the signature of the `Documents.import_` method to allow passing any object of type `Iterable` as the `documents` argument and allows an additional keyword argument called `batch_size`.

## PR Checklist

- [X] I have read and signed the [Contributor License Agreement](https://forms.gle/PZyiY5N2GDQU8GsV9).

### CODE REVIEW

File: typesense/documents.py [hunk 1]
```diff
@@ -50,19 +50,33 @@ def import_jsonl(self, documents_jsonl):
...
+                response_objs = []
+                batch = []
+                for document in documents:
+                    batch.append(document)
+                    if (len(batch) == batch_size):
+                        api_response = self.import_(batch, params)
+                        response_objs.extend(api_response)
+                        batch = []
+                if batch:
+                    self.import_(batch, params)
```
Comment (kishorenc) [inline typesense/documents.py:65]: Should be:

```
api_response = self.import_(batch, params)
```

Correct?

**Review by kishorenc — CHANGES_REQUESTED** [review]

File: typesense/documents.py (hunk 1 above)
Comment (pratheekrebala) [reply typesense/documents.py]: Yep! I just fixed in 3c2f708

**Comment by kishorenc** [comment]
Sorry, there's now a conflict because of the changes from https://github.com/typesense/typesense-python/pull/26/files that needs resolution.

**Comment by pratheekrebala** [comment]
Resolved!

**Comment by pratheekrebala** [comment]
Resolved!

**Review by kishorenc — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**